File: app/src/main/java/com/krev/trycrypt/activity/PlacesActivity.java
package com.krev.trycrypt.activity;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.widget.ListView;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.krev.trycrypt.R;
import com.krev.trycrypt.adapters.GooglePlaceAdapter;

public class PlacesActivity extends AppCompatActivity {

    private static final String TAG = PlacesActivity.class.toString();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_places);
        Log.d(TAG, "onCreate: " + "uhmm, something is ok, buuuuut");
        GooglePlaceAdapter googlePlaceAdapter = new GooglePlaceAdapter(MapActivity.googlePlace, this);
        try {
            Log.d(TAG, "onCreate: " + new ObjectMapper().writeValueAsString(MapActivity.googlePlace));
        } catch (JsonProcessingException e) {
            e.printStackTrace();
        }
        ListView listView = ((ListView) findViewById(R.id.listViewPlaces));
        listView.setAdapter(googlePlaceAdapter);
        registerForContextMenu(listView);
    }
}

File: app/src/main/java/com/krev/trycrypt/activity/PlacesActivity.kt
package com.krev.trycrypt.activity

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log
import android.widget.ListView
import com.baoyz.widget.PullRefreshLayout
import com.krev.trycrypt.R
import com.krev.trycrypt.adapter.PlacesAdapter
import com.krev.trycrypt.model.Config
import com.krev.trycrypt.server.UserController
import com.krev.trycrypt.util.DrawerUtils

class PlacesActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        Config.activity = this
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_places)
        DrawerUtils.getDrawer(this)
        val places = PlacesAdapter().add(Config.places)
        (findViewById(R.id.listViewPlaces) as ListView).adapter = places

        val layout = findViewById(R.id.places_swipeRefreshLayout) as PullRefreshLayout

        layout.setOnRefreshListener {
            UserController.places().thenApplyAsync {
                runOnUiThread {
                    val given = it.toHashSet()
                    Log.d("PlacesActivity", "updating with delta ${places.count - given.count()})")
                    places.clear(given)
                    Config.places = given
                    layout.setRefreshing(false)
                }
            }
        }
    }
}
